src/rgw/rgw_lc_tier.cc

```diff
@@ -920,50 +920,48 @@ class RGWLCStreamObjToCloudMultipartCR : public RGWCoroutine {
 
   map<string, string> new_attrs;
 
-  int ret_err{0};
-
   rgw_raw_obj status_obj;
   bufferlist bl;
 
+  rgw_lc_obj_properties obj_properties;
+  RGWBucketInfo b;
+  string target_obj_name;
+  rgw_bucket target_bucket;
+  int ret = 0;
+
   public:
-  RGWLCStreamObjToCloudMultipartCR(RGWLCCloudTierCtx& _tier_ctx) : RGWCoroutine(_tier_ctx.cct),  tier_ctx(_tier_ctx) {}
+  RGWLCStreamObjToCloudMultipartCR(RGWLCCloudTierCtx& _tier_ctx)
+    : RGWCoroutine(_tier_ctx.cct),  tier_ctx(_tier_ctx),
+          obj_properties(tier_ctx.o.meta.mtime, tier_ctx.o.meta.etag,
+                         tier_ctx.o.versioned_epoch, tier_ctx.acl_mappings,
+                         tier_ctx.target_storage_class){}
 
   int operate() override {
-    rgw_lc_obj_properties obj_properties(tier_ctx.o.meta.mtime,
-        tier_ctx.o.meta.etag,
-        tier_ctx.o.versioned_epoch,
-        tier_ctx.acl_mappings,
-        tier_ctx.target_storage_class);
+    reenter(this) {
 
-    //rgw_obj& obj = (*tier_ctx.obj)->get_obj();
-    obj_size = tier_ctx.o.meta.size;
+      obj_size = tier_ctx.o.meta.size;
 
-    rgw_bucket target_bucket;
-    target_bucket.name = tier_ctx.target_bucket_name;
+      target_bucket.name = tier_ctx.target_bucket_name;
 
-    string target_obj_name;
-    target_obj_name = tier_ctx.bucket_info.bucket.name + "/" +
-                      (*tier_ctx.obj)->get_name();
-    if (!tier_ctx.o.is_current()) {
-      target_obj_name += get_key_instance((*tier_ctx.obj)->get_key());
-    }
-    rgw_obj dest_obj(target_bucket, target_obj_name);
-    rgw_rest_obj rest_obj;
-
-    reenter(this) {
+      target_obj_name = tier_ctx.bucket_info.bucket.name + "/" +
+                        (*tier_ctx.obj)->get_name();
+      if (!tier_ctx.o.is_current()) {
+        target_obj_name += get_key_instance((*tier_ctx.obj)->get_key());
+      }
+      dest_obj.init(target_bucket, target_obj_name);
 
-    status_obj = rgw_raw_obj(tier_ctx.store->get_zone()->get_params().log_pool,
-        "lc_multipart_" + (*tier_ctx.obj)->get_oid());
+      status_obj = rgw_raw_obj(tier_ctx.store->get_zone()->get_params().log_pool,
+          "lc_multipart_" + (*tier_ctx.obj)->get_oid());
 
-      ret_err = tier_ctx.store->get_system_obj(tier_ctx.dpp, status_obj.pool,
+      ret = tier_ctx.store->get_system_obj(tier_ctx.dpp, status_obj.pool,
```

Comment:
this coroutine is calling `get_system_obj()` `put_system_obj()` and `delete_system_obj()`, all of which are synchronous. you'll want to call the async versions instead: `RGWSimpleRadosReadCR`, `RGWSimpleRadosWriteCR`, and `RGWRadosRemoveCR`

Reply:
Initially had used RGWSimpleRados*CR but there is no abstraction for RGWAsyncRadosProcessor made part of Zipper 
 yet and hence couldn't use it directly. 

@dang mentioned that this may be addressed when sysobj is replaced in Zipper.

Reply:
So until the APIs are available in common sal layer, do you suggest to explicitly typecast to RGWRadosStore and fetch async_rados() for now?

Reply:
i don't think `RGWAsyncRadosProcessor` itself has any dependency on rados, it's just a thread pool that coroutines use to dispatch synchronous work, while the coroutines are themselves asynchronous (they yield until the thread pool passes back the result)

that said, using `RGWAsyncRadosProcessor` is a hack that just avoids having to actually make the logic asynchronous, and i'd much rather we kill that thread pool than reify it as a part of zipper

in the meantime, i think it's okay for this tiering to be rados-backend-only and require a `RGWRadosStore` instead of the generic `RGWStore`; we just have the pick the right place in lifecycle processing to try `dynamic_cast<RGWRadosStore>` and return `ENOTSUP` on null

Reply:
okay sure thanks. Will make the changes.